polygeist/lib/Dialect/Polygeist/Analysis/MemoryAccessAnalysis.cpp

```diff
@@ -88,8 +88,8 @@ static SmallVector<Value> computeThreadVector(FunctionOpInterface funcOp,
 
   // Return the index value of an operation.
   auto getIndexValue = [&](T &op) -> std::optional<APInt> {
-    std::optional<TypedValue<IntegerType>> idx = op.getIndex();
-    return idx.has_value() ? getConstIntegerValue(*idx, solver) : APInt();
+    TypedValue<IntegerType> idx = op.getIndex();
```

Comment:
The original code crashes, as `getIndex ` always return a value, but could be nulloptr.